### PR TITLE
Create the base classes for the chain of responsibilities for Handlers

### PR DESCRIPTION
**Pull request by saurabh500**
Create the base classes for the chain of responsibilities for Handlers.

### CODE REVIEW

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/IO/TdsStream.cs [hunk 1]
```diff
@@ -0,0 +1,181 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Data.SqlClientX.IO
```
Comment (benrr101) [inline src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/IO/TdsStream.cs]: I don't think this is meant to go in this PR

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/IO/TdsStreamPacketType.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿namespace Microsoft.Data.SqlClientX.IO
```
Comment (benrr101) [inline src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/IO/TdsStreamPacketType.cs]: Likewise, this shouldn't be in this PR?

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/Handler.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient;
+using Microsoft.Data.SqlClient.SNI;
+using Microsoft.Data.SqlClientX.IO;
+
+
+namespace Microsoft.Data.SqlClientX.Handlers
+{
+    internal interface IHandler<TRequest> where TRequest : HandlerRequest
```
Comment (benrr101) [inline src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/Handler.cs:16]: Eventually these should be separate files, but to unblock for now, we can have them in one file

Comment (benrr101) [reply src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/Handler.cs:16]: Likewise, there's some style changes that I'd push for

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/Handler.cs [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    {
+        /// <summary>
+        /// Stores the internal handler to be executed next.
+        /// </summary>
+        private IHandler<TRequest> _nextHandler;
+
+        /// <summary>
+        /// Property for interacting with the Next Handler.
+        /// </summary>
+        public IHandler<TRequest> NextHandler { get => _nextHandler; set => _nextHandler = value; }
```
Comment (benrr101) [inline src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/Handler.cs:50]: Shouldn't this just be an auto-property?

**Review by benrr101 — APPROVED** [review]
These aren't blocking changes, but I really will push to get these changes made at *some* point

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/Handler.cs (hunk 3 above)
Comment (saurabh500) [reply src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/Handler.cs:16]: I can take care of all this now. Tell me what I am missing ?

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/Handler.cs (hunk 4 above)
Comment (saurabh500) [reply src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/Handler.cs:50]: Actually I think this should be virtual. The base can provide an auto-property and the subclasses can do more complex stuff .

**Review by benrr101 — APPROVED** [review]
As per @saurabh500's request, here's the style fix requests :)

**Review by cheenamalhotra — APPROVED** [review]

**Pull request merged**